plugins/backstage-plugin-coder/docs/guides/coder-api.md

```diff
@@ -0,0 +1,257 @@
+
+## Authentication
+
+All API calls to **any** of the Coder API functions will fail if you have not authenticated yet. Authentication can be handled via any of the official Coder components that can be imported via the plugin. However, if there are no Coder components on the screen, the `CoderProvider` component will automatically\* inject a fallback auth button for letting the user add their auth info.
+
+<-- Add video of auth flow with fallback button -->
+
+Once the user has been authenticated, all Coder API functions will become available. When the user unlinks their auth token (effectively logging out), all cached queries that start with `CODER_QUERY_KEY_PREFIX` will automatically be vacated.
+
+\* This behavior can be disabled. Please see our [advanced API guide](./coder-api-advanced.md) for more information.
+
+## Connecting a custom query client to the Coder plugin
+
+By default, the Coder plugin uses and manages its own query client. This works perfectly well if you aren't using React Query for any other purposes, but if you are using it throughout your Backstage deployment, it can cause issues around redundant state (e.g., not all cached data being vacated when the user logs out).
+
+To prevent this, you will need to do two things:
+
+1. Pass in your custom React Query query client into the `CoderProvider` component
+2. "Group" your queries with the Coder query key prefix
+
+### Passing in a custom query client
+
+The `CoderProvider` component accepts an optional `queryClient` prop. When provided, the Coder plugin will use this client for **all** queries (those made by the built-in Coder components, or any custom components that you put inside `CoderProvider`).
+
+```tsx
+const customQueryClient = new QueryClient();
+
+<CoderProvider queryClient={customQueryClient}>
+  <YourCustomComponentsThatNeedAccessToTheCoderPlugin />
+</CoderProvider>;
+```
+
+### Grouping queries with the Coder query key prefix
+
+The plugin exposes a `CODER_QUERY_KEY_PREFIX` constant that you can use to group all Coder queries together for `useQuery` and `useQueryClient`. All queries made by official Coder components put this as the first value of their query key. The `useCoderQuery` convenience hook also automatically injects this constant at the beginning of all query keys (even if not explicitly added).
+
+```tsx
+// Starting all query keys with the constant "groups" them together
+const coderApi = useCoderApi();
+const workspacesQuery = useQuery({
+  queryKey: [CODER_QUERY_KEY_PREFIX, 'workspaces'],
+  queryFn: () =>
+    coderApi.getWorkspaces({
+      limit: 10,
+    }),
+});
+
+const workspacesQuery2 = useCoderQuery({
+  // The query key will automatically have CODER_QUERY_KEY_PREFIX added to the
+  // beginning
+  queryKey: ['workspaces'],
+  queryFn: ({ coderApi }) =>
+    coderApi.getWorkspaces({
+      limit: 10,
+    }),
+});
```

Comment:
I think the first two examples here are unnecessary given the preceding paragraph.
The next two examples are helpful in defining what 'group' means in this context.